libs/src/services/ethContracts/governanceClient.js

```diff
@@ -105,6 +136,160 @@ class GovernanceClient extends ContractClient {
     )
     return this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT)
   }
+
+  async getProposalById (
+    id
+  ) {
+    const method = await this.getMethod(
+      'getProposalById',
+      id
+    )
+    const proposal = await method.call()
+    const formattedProposal = formatProposal(proposal)
+    return formattedProposal
+  }
+
+  async getProposalDescriptionById (
+    id
+  ) {
+    const method = await this.getMethod(
+      'getProposalDescriptionById',
+      id
+    )
+    return method.call()
+  }
+
+  async getInProgressProposals () {
+    const method = await this.getMethod('getInProgressProposals')
+    const ids = await method.call()
+    return ids
+  }
+
+  async submitProposal ({
+    targetContractRegistryKey,
+    callValue,
+    functionSignature,
+    callData,
+    description
+  }) {
+    const method = await this.getMethod(
+      'submitProposal',
+      targetContractRegistryKey,
+      callValue,
+      functionSignature,
+      callData,
+      description
+    )
+    const tx = await this.web3Manager.sendTransaction(method, DEFAULT_GAS_AMOUNT * 10)
+    if (tx && tx.events && tx.events.ProposalSubmitted && tx.events.ProposalSubmitted.returnValues) {
+      const id = tx.events.ProposalSubmitted.returnValues.proposalId
+      return id
+    }
+    throw new Error('No proposal Id')
```

Comment:
dont know all the error states, but just wanna double check that this doens't get thrown if the proposal is just malformed or something

Reply:
yeah fair point, will change the error message